File: DaS.ScriptLib/Game/Data/Structures/BossFightInfo.vb
Namespace Game.Data.Structures
    Public Class BossFightInfo
        Public ReadOnly Property SpawnFlag As EventFlag.Boss = EventFlag.Boss.AsylumDemon
        Public ReadOnly Property Name As String = "?BossName?"

        Public PlayerWarp As Loc = Loc.Zero
        Public World As Integer = 0
        Public Area As Integer = 0
        Public WarpID As Integer = -1
        Public PlayerAnim As Integer = 7410
        Public AdditionalFlags As Integer() = {}
        Public EntranceLua As String = ""

        Public Sub New(spawnFlag As EventFlag.Boss, name As String)
            Me.SpawnFlag = spawnFlag
            Me.Name = name
        End Sub

    End Class
End Namespace

File: DaS.ScriptLib.NEW/Game/Data/Structures/BossFightInfo.cs
namespace DaS.ScriptLib.Game.Data.Structures
{
    public class BossFightInfo
	{
		public EventFlag.Boss SpawnFlag { get; private set;  }
		public string Name { get; private set; }

		public Loc PlayerWarp = Loc.Zero;
		public int World = 0;
		public int Area = 0;
		public int WarpID = -1;
		public int PlayerAnim = 7410;
		public int[] AdditionalFlags = {
			
		};

		public string EntranceLua = "";
		public BossFightInfo(EventFlag.Boss spawnFlag, string name)
		{
			this.SpawnFlag = spawnFlag;
			Name = name;
		}

	}
}
